src/server/pkg/obj/minio_client.go

```diff
@@ -23,16 +24,43 @@ func newMinioClient(endpoint, bucket, id, secret string, secure bool) (*minioCli
 	}, nil
 }
 
-func (c *minioClient) Writer(name string) (io.WriteCloser, error) {
+// Represents minio writer structure with pipe and the error channel
+type minioWriter struct {
+	errChan chan error
+	pipe    *io.PipeWriter
+}
+
+// Creates a new minio writer and a go routine to upload objects to minio server
+func newMinioWriter(client *minioClient, name string) *minioWriter {
 	reader, writer := io.Pipe()
-	go func(reader *io.PipeReader) {
-		_, err := c.PutObject(c.bucket, name, reader, "application/octet-stream")
+	w := &minioWriter{
+		errChan: make(chan error),
+		pipe:    writer,
+	}
+	go func() {
+		_, err := client.PutObject(client.bucket, name, reader, "application/octet-stream")
 		if err != nil {
 			reader.CloseWithError(err)
-			return
 		}
-	}(reader)
-	return writer, nil
+		w.errChan <- err
+	}()
+	return w
+}
+
+func (w *minioWriter) Write(p []byte) (int, error) {
+	return w.pipe.Write(p)
+}
+
+// This will block till upload is done
+func (w *minioWriter) Close() error {
+	if err := w.pipe.Close(); err != nil {
+		return err
```

Comment:
I could be missing something, but I think if we return the error here, then we won't get to the channel receive below, in which case the goro in `newMinioWriter` might forever block on sending.